(DOCSP-12990): Add missing parts of query engine page (#633)
mongodb/docs-realm: source/ios/query-engine.txt

<https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/Predicates/AdditionalChapters/Introduction.html>`__.

Predicates
----------

You can build a filter with NSPredicate:

.. tabs-realm-languages::

   .. tab::
      :tabid: swift

      .. literalinclude:: /examples/generated/code/start/QueryEngine.codeblock.predicates.swift
         :language: swift

   .. tab::
      :tabid: objective-c

      .. literalinclude:: /examples/generated/code/start/QueryEngine.codeblock.predicates.m
         :language: objective-c

Expressions
~~~~~~~~~~~

Filters consist of **expressions** in an NSPredicate. An expression consists of
one of the following:

- The name of a property of the object currently being evaluated.
- An operator and up to two argument expression(s).
- A literal string, number, or date.

Dot Notation
~~~~~~~~~~~~

> When referring to an object property, you can use the **dot notation** to refer

Reviewer: I feel like the article isn't necessary here. What do you think?
```suggestion
When referring to an object property, you can use **dot notation** to refer
```

   .. tabs-realm-languages::

      .. tab::
         :tabid: swift

         .. literalinclude:: /examples/generated/code/start/QueryEngine.codeblock.models.swift
             :language: swift

      .. tab::
         :tabid: objective-c

         .. literalinclude:: /examples/generated/code/start/QueryEngine.codeblock.models.m
             :language: objective-c

>    You can set up the realm for these examples as follows:

Reviewer: ```suggestion
   You can set up the {+realm+} for these examples as follows:
```

Reviewer: "as follows" is a little vague. Could we change this to "with the following Realm object models" or something like that?

     - Evaluates to ``true`` if the left-hand numerical or date expression is greater than or equal to the right-hand numerical or date expression. For dates, this evaluates to ``true`` if the left-hand date is later than or the same as the right-hand date.

   * - ``in``

>      - Evaluates to ``true`` if the left-hand expression is in the right-hand list or string.

Reviewer: do you mean:
```suggestion
     - Evaluates to ``true`` if the left-hand expression is in the right-hand list of strings.
```
?

Author: No. I mean the property can be a list or a string (but not, say, a number). Open to suggestions.

Reviewer: No problem, seems like i misunderstood what was going on here.
